File: BMWCenter/Core/Care/SeverityRouter.swift
import Foundation

enum SeverityRouter {
    static func plan(for severity: CueSeverity, appInBackground: Bool = false) -> CareChannelPlan {
        switch severity {
        case .critical:
            return CareChannelPlan(
                speak: true,
                carPlayRow: true,
                carPlayAlert: true,
                phoneChip: true,
                fullScreen: true,
                notification: true,
                haptic: "error",
                toneCount: 2
            )
        case .protective:
            return CareChannelPlan(
                speak: true,
                carPlayRow: true,
                carPlayAlert: false,
                phoneChip: true,
                fullScreen: false,
                notification: appInBackground,
                haptic: "warning",
                toneCount: 1
            )
        case .coach:
            return CareChannelPlan(
                speak: true,
                carPlayRow: true,
                carPlayAlert: false,
                phoneChip: true,
                fullScreen: false,
                notification: false,
                haptic: nil,
                toneCount: 0
            )
        case .celebration:
            return CareChannelPlan(
                speak: true,
                carPlayRow: true,
                carPlayAlert: false,
                phoneChip: true,
                fullScreen: false,
                notification: false,
                haptic: "success",
                toneCount: 1
            )
        }
    }

    static func alertSeverity(from cue: CueSeverity) -> AlertSeverity {
        switch cue {
        case .critical: return .critical
        case .protective: return .warning
        case .coach, .celebration: return .info
        }
    }
}

File: mobile/src/core/care/severityRouter.ts
import type { AlertSeverity } from "../alerts/alertTypes";
import type { CueSeverity, CareChannelPlan } from "./careTypes";

export function planFor(severity: CueSeverity, appInBackground = false): CareChannelPlan {
  switch (severity) {
    case "critical":
      return { speak: true, phoneChip: true, fullScreen: true, notification: true, toneCount: 2 };
    case "protective":
      return { speak: true, phoneChip: true, fullScreen: false, notification: appInBackground, toneCount: 1 };
    case "coach":
      return { speak: true, phoneChip: true, fullScreen: false, notification: false, toneCount: 0 };
    case "celebration":
      return { speak: true, phoneChip: false, fullScreen: false, notification: false, toneCount: 1 };
  }
}

export function alertSeverityFrom(cue: CueSeverity): AlertSeverity {
  switch (cue) {
    case "critical":
      return "critical";
    case "protective":
      return "warning";
    case "coach":
    case "celebration":
      return "info";
  }
}
